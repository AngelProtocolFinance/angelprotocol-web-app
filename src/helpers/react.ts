import { forwardRef } from "react";

/** wraps forwardRef to preserve generic */
export function fixedForwardRef<T, P = {}>(
  render: (props: P, ref: React.Ref<T>) => React.ReactNode
): (props: P & React.RefAttributes<T>) => React.ReactNode {
  return forwardRef(render) as any;
}
