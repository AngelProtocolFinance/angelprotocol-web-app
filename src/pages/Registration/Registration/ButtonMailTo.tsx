type Props = { mailTo: string; label: string };

export default function ButtonMailTo({ mailTo, label }: Props) {
  return (
    <a
      href={`mailto:${mailTo}?subject=${encodeURIComponent(
        "Charity Registration: Trouble with confirmation email"
      )}`}
      className="italic underline"
    >
      {label}
    </a>
  );
}
