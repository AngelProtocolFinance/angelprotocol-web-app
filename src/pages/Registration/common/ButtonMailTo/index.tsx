type Props = { label: string; mailTo: string; subject: string };

export function ButtonMailTo({ mailTo, label, subject }: Props) {
  return (
    <a
      href={`mailto:${mailTo}${
        subject ? `?subject=${encodeURIComponent(subject)}` : ""
      }`}
      className="underline text-xs text-orange"
    >
      {label}
    </a>
  );
}
