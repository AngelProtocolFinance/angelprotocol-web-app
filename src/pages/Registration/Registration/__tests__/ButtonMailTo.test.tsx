import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import ButtonMailTo from "../ButtonMailTo";

type Props = { label: string; mailTo: string; subject: string };

const MockComponent = (props: Props) => {
  return (
    <MemoryRouter>
      <ButtonMailTo {...props} />
    </MemoryRouter>
  );
};

describe("ButtonMailTo tests", () => {
  it("matches snapshot", () => {
    const component = renderer.create(
      <MockComponent label="Label" mailTo="some@email.com" subject="Subject" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("opens email with correct recepient and subject", () => {
    const expectedLabel = "Label";
    const email = "test@email.com";
    render(
      <MockComponent
        label={expectedLabel}
        mailTo={email}
        subject="Charity Registration: Trouble with confirmation email"
      />
    );

    const link = screen.getByRole("link");
    expect(link).toHaveAttribute(
      "href",
      `mailto:${email}?subject=Charity%20Registration%3A%20Trouble%20with%20confirmation%20email`
    );
  });
});
