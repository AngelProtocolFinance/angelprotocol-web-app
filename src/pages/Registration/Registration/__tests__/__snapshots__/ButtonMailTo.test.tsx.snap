// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonMailTo tests matches snapshot 1`] = `
<a
  className="italic underline"
  href="mailto:some@email.com?subject=Charity%20Registration%3A%20Trouble%20with%20confirmation%20email"
>
  Label
</a>
`;
